src/status_im/ui/screens/discover/views.cljs

```diff
@@ -1,95 +1,114 @@
+    [status-im.i18n :as i18n]
+    [status-im.ui.screens.discover.styles :as styles]
+    [status-im.ui.screens.contacts.styles :as contacts-st]))
 
 (defn get-hashtags [status]
   (let [hashtags (map #(str/lower-case (str/replace % #"#" "")) (re-seq #"[^ !?,;:.]+" status))]
     (or hashtags [])))
 
 (defn toolbar-view [show-search? search-text]
-  [toolbar-with-search
+  [toolbar/toolbar-with-search
    {:show-search?       show-search?
     :search-text        search-text
     :search-key         :discover
-    :title              (label :t/discover)
-    :search-placeholder (label :t/search-tags)
+    :title              (i18n/label :t/discover)
+    :search-placeholder (i18n/label :t/search-tags)
     :nav-action         (act/hamburger drawer/open-drawer!)
     :on-search-submit   (fn [text]
                           (when-not (str/blank? text)
                             (let [hashtags (get-hashtags text)]
                               (dispatch [:set :discover-search-tags hashtags])
                               (dispatch [:navigate-to :discover-search-results]))))}])
 
-(defn title [label-kw spacing?]
-  [view st/section-spacing
-   [text {:style      (merge (get-in platform-specific [:component-styles :discover :subtitle])
-                             (when spacing? {:margin-top 16}))
-          :uppercase? (get-in platform-specific [:discover :uppercase-subtitles?])
-          :font       :medium}
-    (label label-kw)]])
 
-(defview discover-popular [{:keys [contacts current-account]}]
+(defview top-status-for-popular-hashtag [{:keys [tag current-account]}]
+  (letsubs [discoveries [:get-popular-discoveries 1 [tag]]]
+    [view (merge styles/popular-list-container
+                 (get-in platform/platform-specific [:component-styles :discover :popular]))
+     [view styles/row
+      [view {}
+       [touchable-highlight {:on-press #(do (dispatch [:set :discover-search-tags [tag]])
+                                            (dispatch [:navigate-to :discover-search-results]))}
+        [view {}
+         [text {:style styles/tag-name
+                :font  :medium}
+          (str " #" (name tag))]]]]
+      [view styles/tag-count-container
+       [text {:style styles/tag-count
+              :font  :default}
+        (:total discoveries)]]]
+     [components/discover-list-item {:message         (first (:discoveries discoveries))
+                                     :show-separator? false
+                                     :current-account current-account}]]))
+
+(defview popular-hashtags-preview [{:keys [contacts current-account]}]
   (letsubs [popular-tags [:get-popular-tags 10]]
-    [view st/popular-container
-     [title :t/popular-tags false]
+    [view styles/popular-container
+     [components/title :t/popular-tags false :t/all #(do (dispatch [:set :discover-search-tags (map :name popular-tags)])
+                                                         (dispatch [:navigate-to :discover-all-hashtags]))]
```

Comment:
Same as above, please add `TODO` comment that we need to refactor it in the very near future

Reply:
Makes sense!